bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -4362,6 +4478,628 @@ void ReplicaImp::setConflictDetectionBlockId(const ClientRequestMsg &clientReqMs
   execReq.requestSize = requestSize;
 }
 
+// TODO(GG): understand and handle the update in
+// https://github.com/vmware/concord-bft/commit/2d812aa53d3821e365809774afb30468134c026d
+// TODO(GG): When the replica starts, this method should be called before we start receiving messages ( because we may
+// have some committed seq numbers)
+// TODO(GG): move the "logic related to requestMissingInfo" to the caller of this method (e.g., by returning a value
+// to the caller)
+void ReplicaImp::tryToStartOrFinishExecuting(const bool requestMissingInfo) {
+  ConcordAssert(!isCollectingState());
+  ConcordAssert(currentViewIsActive());
+  ConcordAssertGE(lastExecutedSeqNum, lastStableSeqNum);
+
+  if (activeExecutions_ > 0) {
+    ConcordAssert(startedToExecute);
+    return;  // because this method will be called again as soon as the execution completes
+  }
+
+  if (lastExecutedSeqNum >= lastStableSeqNum + kWorkWindowSize) {
+    if (startedToExecute) {
+      startedToExecute = false;
+      onFinishtExecuting();
+    }
+
+    return;
+  }
+
+  const SeqNum nextExecutedSeqNum = lastExecutedSeqNum + 1;
+  const SeqNumInfo &seqNumInfo = mainLog->get(nextExecutedSeqNum);
+  PrePrepareMsg *prePrepareMsg = seqNumInfo.getPrePrepareMsg();
+
+  const bool ready = (prePrepareMsg != nullptr) && (seqNumInfo.isCommitted__gg());
+
+  if (!ready) {
+    if (startedToExecute) {
+      startedToExecute = false;
+      onFinishtExecuting();
+    } else {
+      if (requestMissingInfo) {
```

Comment:
I think that request missing info sould happen in any case

Reply:
Done.